include/motor-control/core/motor.hpp

```diff
@@ -2,42 +2,41 @@
 
 #include <variant>
 
-#include "common/core/message_queue.hpp"
+#include "common/core/freertos_message_queue.hpp"
 #include "linear_motion_system.hpp"
 #include "motion_controller.hpp"
 #include "motor_driver.hpp"
 #include "motor_driver_config.hpp"
 #include "motor_messages.hpp"
 #include "spi.hpp"
 
+namespace motor_hardware {
+class MotorHardwareIface;
+};
+
 namespace motor_class {
 
 using namespace motor_messages;
 using namespace motor_driver_config;
+using namespace freertos_message_queue;
 
-template <spi::TMC2130Spi SpiDriver, template <class> class PendingQueueImpl,
-          template <class> class CompletedQueueImpl,
-          lms::MotorMechanicalConfig MEConfig>
-requires MessageQueue<PendingQueueImpl<Move>, Move> &&
-    MessageQueue<CompletedQueueImpl<Ack>, Ack>
+template <spi::TMC2130Spi SpiDriver, lms::MotorMechanicalConfig MEConfig>
 struct Motor {
-    using GenericQueue = PendingQueueImpl<Move>;
-    using CompletedQueue = CompletedQueueImpl<Ack>;
+    using GenericQueue = FreeRTOSMessageQueue<Move>;
+    using CompletedQueue = FreeRTOSMessageQueue<Ack>;
     Motor(SpiDriver& spi, lms::LinearMotionSystemConfig<MEConfig> lms_config,
-          motion_controller::HardwareConfig& config,
+          motor_hardware::MotorHardwareIface& hardware_iface,
           MotionConstraints constraints, RegisterConfig driver_config,
           GenericQueue& queue, CompletedQueue& completed_queue)
         : pending_move_queue(queue),
```

Comment:
We can probably not make `pending_move_queue` and `completed_move_queue` members of the Motor struct

Reply:
Actually those queue refs are used by the move group executor to get a way to talk to the motor interrupt handler - the `Motor` class is a big bag of references that link the motor system together.